app/src/main/java/org/kinecosystem/kinit/view/earn/WebTaskTruexFragment.kt

```diff
@@ -16,6 +18,18 @@ import org.kinecosystem.kinit.viewmodel.earn.WebFragmentActions
 import org.kinecosystem.kinit.viewmodel.earn.WebTaskTruexViewModel
 
 class WebTaskTruexFragment : BaseFragment(), WebFragmentActions {
+    override fun openBrowser(url: String) {
+        val parsedUrl = Uri.parse(url)
+        val intent = Intent(Intent.ACTION_VIEW, parsedUrl)
+        context?.let {
+            if (parsedUrl != null || intent.resolveActivity(it.packageManager) != null) {
+                showErrorDialog()
+            } else {
+                startActivity(intent)
+            }
+        }
```

Comment:
Sorry in the above code, you may crash if parsedUrl is null
Here is a way to write this correctly
```
context?.let {
   val parsedUrl = Uri.parse(url)
   val intent = if (parsedUrl != null) Intent(Intent.ACTION_VIEW, parsedUrl) else null
   if (intent?.resolveActivity(context.packageManager) != null ){
      startActivity(intent)
   }
  else {
     showErrorDialog()
  }   
}
```